src/net/mod.rs

```diff
@@ -1,41 +1,14 @@
 //! Networking primitives
 //!
-pub mod tcp;
-pub mod udp;
-
-use {poll, Poll};
-use std::io;
-use std::sync::atomic::{AtomicUsize, Ordering};
-
-/// Used to associate an IO type with a Selector
-#[derive(Debug)]
-struct SelectorId {
-    id: AtomicUsize,
-}
-
-impl SelectorId {
-    fn new() -> SelectorId {
-        SelectorId {
-            id: AtomicUsize::new(0),
-        }
-    }
-
-    fn associate_selector(&self, poll: &Poll) -> io::Result<()> {
-        let selector_id = self.id.load(Ordering::SeqCst);
+//! The types provided in this module are non-blocking by default and are
+//! designed to be portable across all supported Mio platforms. As long as the
+//! [portability guidelines] are followed, the behavior should be identical no
+//! matter the target platform.
+//!
+//! [portability guidelines]: ../struct.Poll.html#portability
 
-        if selector_id != 0 && selector_id != poll::selector(poll).id() {
-            Err(io::Error::new(io::ErrorKind::Other, "socket already registered"))
-        } else {
-            self.id.store(poll::selector(poll).id(), Ordering::SeqCst);
-            Ok(())
-        }
-    }
-}
+mod tcp;
+mod udp;
 
-impl Clone for SelectorId {
-    fn clone(&self) -> SelectorId {
-        SelectorId {
-            id: AtomicUsize::new(self.id.load(Ordering::SeqCst)),
-        }
-    }
-}
+pub use self::tcp::{TcpListener, TcpStream, Shutdown};
```

Comment:
Perhaps `Shutdown` could be left out? (I think it's just a libstd reexport, right?)